### ISSUE TITLE
Semantic Annotations - Inconsistencies

### ISSUE THREAD

**Issue by StephanEwen** [stratosphere/stratosphere#846]
This is basically a thread for discussion. There are still a few inconsistencies in the semantic annotations. The following examples illustrate that:

_Constant fields_:
`"1 -> 0,1" "3->4"`  means field 1 is copied unmodified to 0 and 1, while 3 becomes 4.

`"1 ->0,1 ; 3->4"`   means the same thing, expressed in a single string (semicolon is the delimiter between statements about fields).

`"1 , 3"` means that 1 and 3 remain constant as 1 and 3. Note that comma is the delimiter here.

_Read Fields_:
`"0 , 2"` means that fields 0 and 2 are read. Note that here, the delimiter is the comma, like with the standalone constant fields, unlike the constant fields with arrow notation.

I find that a bit inconsistent, especially the mixed use of comma and semicolon for the constant fields.
- Do we want to keep it that way?
- Or use the  semicolon for the standalone constant fields and read fields?
- Or use the comma in the constant fields (finding an alternative delimiter for the right hand side of the arrow?

**Comment by markus-h** [comment]
For me a semicolon for indicating that fields are staying constant would be more intuitive. In that way "1 ; 3" would just be a shortcut for "1->1 ; 3->3"

**Comment by fhueske** [comment]
I agree, the semicolon should basically separate different constant annotations (each valid for exactly one field of the input) that can also be put into individual strings, i.e.,
`@ConstantFields("1->2,3; 4; 5->7")`
should be equivalent to
`@ConstantFields("1->2,3", "4", "5->7")`
